### PR TITLE
json: use isoformat

### PR DESCRIPTION
**Pull request by ThomasWaldmann**
a first step for #2376, see commit comments.

### CODE REVIEW

File: src/borg/helpers.py [hunk 1]
```diff
@@ -1781,11 +1802,6 @@ def __init__(self, archive, format, *, json_lines=False):
         for hash_function in hashlib.algorithms_guaranteed:
             self.add_key(hash_function, partial(self.hash_item, hash_function))
         self.used_call_keys = set(self.call_keys) & self.format_keys
-        if self.json_lines:
-            self.item_data = {}
-            self.format_item = self.format_item_json
-        else:
-            self.item_data = static_keys
```
Comment (ThomasWaldmann) [inline src/borg/helpers.py]: needed to move this upwards, so the time formatting gets set up before it is used.

**Comment by ThomasWaldmann** [comment]
https://travis-ci.org/borgbackup/borg/jobs/261426845 wat?

File: src/borg/helpers.py [hunk 2]
```diff
@@ -717,11 +717,17 @@ def safe_timestamp(item_timestamp_ns):
...
-    """use ISO-8601 date and time format
+    """use ISO-8601-like date and time format (human readable, with wkday and blank date/time separator)
     """
     return t.strftime('%a, %Y-%m-%d %H:%M:%S')
 
 
+def isoformat_time(t):
+    """use ISO-8601 date and time format (machine readable, no wkday, no microseconds either)
+    """
+    return t.strftime('%Y-%m-%dT%H:%M:%S')  # note: first make all datetime objects tz aware before adding %z here.
```
Comment (enkore) [inline src/borg/helpers.py]: Could t.isoformat() be used here, or does it something wrong (with TZ or so)?

**Comment by ThomasWaldmann** [comment]
added tests, collapsed a typo fix. merge?

**Comment by ThomasWaldmann** [comment]
about datetime.isoformat: the new spec param is only available since 3.6 and I guess before it sometimes added the microseconds OR NOT (if == 0). I wanted a specific format, not one depending on the value, thus I used strftime.

**Comment by ThomasWaldmann** [comment]
note: i reset master to the previous cs of this merge. so a new rebase pr of this will need to get merged again.

**Comment by ThomasWaldmann** [comment]
see #2928.

**Pull request merged**